hydromt/raster.py

```diff
@@ -2247,6 +2248,245 @@ def tile_window(shape, px):
+                    "band", drop=True
+                )
+
+            obj_bounds = obj.raster.bounds
+            obj_res = obj.raster.res[0]
+            obj_shape = obj.raster.shape
+
+            t_r = pyproj.Transformer.from_crs(
+                3857, obj.raster.crs.to_epsg(), always_xy=True
+            )
+
+            sd = join(root, f"{zl}")
+            create_folder(sd)
+            txt_path = join(sd, "filelist.txt")
+            file = open(txt_path, "w")
+
+            for transform, bounds, col, row in tile_window(zl, minx, miny, maxx, maxy):
+                ssd = join(sd, f"{col}")
+                create_folder(ssd)
+
+                ul = t_r.transform(bounds[0], bounds[3])
+                lr = t_r.transform(bounds[2], bounds[1])
+                out_bounds = [ul[0], lr[1], lr[0], ul[1]]
+
+                temp_bounds = [
+                    max(out_bounds[0], obj_bounds[0]),
+                    max(out_bounds[1], obj_bounds[1]),
+                    min(out_bounds[2], obj_bounds[2]),
+                    min(out_bounds[3], obj_bounds[3]),
+                ]
+
+                _window = [
+                    abs(x - y) / obj_res for x, y in zip(temp_bounds, obj_bounds)
+                ]
+
+                u = math.floor(round(_window[3], 2))
+                l = math.floor(round(_window[0], 2))
+                h = obj_shape[0] - u - math.floor(round(_window[1], 2))
+                w = obj_shape[1] - l - math.floor(round(_window[2], 2))
+
+                temp = obj[u : u + h, l : l + w]
+
+                tile = np.empty((px_size, px_size), np.float64)
+                tile[:] = nodata
+
+                temp_transform = Affine(
+                    obj_res,
+                    0,
+                    obj_bounds[0] + l * obj_res,
+                    0,
+                    -obj_res,
+                    obj_bounds[3] - u * obj_res,
+                )
+
+                rasterio.warp.reproject(
+                    temp.values,
+                    tile,
+                    src_transform=temp_transform,
+                    src_crs=obj.raster.crs,
+                    dst_transform=transform,
+                    dst_crs="EPSG:3857",
+                    dst_nodata=nodata,
+                    resampling=resampling,
+                )
+
+                temp = xr.DataArray(
+                    name=temp.name,
+                    dims=("y", "x"),
+                    coords={
+                        "y": np.arange(
+                            bounds[3] + 0.5 * transform[4], bounds[1], transform[4]
+                        )[:px_size],
+                        "x": np.arange(
+                            bounds[0] + 0.5 * transform[0],
+                            bounds[2] + 0.5 * transform[0],
+                            transform[0],
+                        )[:px_size],
+                    },
+                    data=tile,
+                )
+
+                del tile
```

Comment:
I'm not 100% sure, but I think that `data=tile`  creates a reference, so the tile might not actually get cleaned up when you call `del tile`. You might need a deep copy here.

Reply:
Haven't had the time to properly investigate this, did implement a deep copy.